src/devices/Ssd1351/Ssd1351.cs

```diff
@@ -0,0 +1,295 @@
+            _spiDevice = spiDevice ?? throw new ArgumentNullException(nameof(spiDevice));
+
+
+            _dcPinId = dataCommandPin;
+            _resetPinId = resetPin;
+
+            if(_gpioDevice == null)
+            {
+                _gpioDevice = new GpioController();
+                _disposeGpioController = true;
+            }
+
+            _gpioDevice.OpenPin(_dcPinId, PinMode.Output);
+            _gpioDevice.OpenPin(_resetPinId, PinMode.Output);
+
+            _spiBufferSize = spiBufferSize;
+
+            SetSegmentReMapColorDepth();
+        }
+
+        /// <summary>
+        /// Convert a color structure to a byte tuple represening the colour in 565 format.
+        /// </summary>
+        /// <param name="color">The color to be converted.</param>
+        private (byte, byte) Color565(Color color)
+        {
+            // get the top 5 MSB of the blue or red value
+            UInt16 retval = (UInt16)((_colorSequence == ColorSequence.ABC ? color.B : color.R) >> 3);
+            // shift right to make room for the green Value
+            retval <<= 6;
+            // combine with the 6 MSB if the green value
+            retval |= (UInt16)(color.G >> 2);
+            // shift right to make room for the red or blue Value
+            retval <<= 5;
+            // combine with the 6 MSB if the red or blue value
+            retval |= (UInt16)((_colorSequence == ColorSequence.ABC ? color.R : color.B) >> 3);
+            return ((byte)(retval >> 8), (byte)(retval & 0xFF));
+        }
+
+        /// <summary>
+        /// Send filled rectangle to the ssd1351 display.
+        /// </summary>
+        /// <param name="color">The color to fill the rectangle with.</param>
+        /// <param name="x">The x co-ordinate of the point to start the rectangle at in pixels (0..126 for the Ssd1351).</param>
+        /// <param name="y">The y co-ordinate of the point to start the rectangle at in pixles (0..126 for the Ssd1351) .</param>
+        /// <param name="w">The width of the rectangle in pixels (1..127 for the Ssd1351).</param>
+        /// <param name="h">The height of the rectangle in pixels (1..127 for the Ssd1351).</param>
+        public void FillRect(Color color, byte x, byte y, byte w, byte h)
+        {
+            Span<byte> colourBytes = stackalloc byte[_colorDepth == ColorDepth.ColourDepth65K ? 2 : 3]; // create a short span that holds the colour data to be sent to the display
+            Span<byte> displayBytes = stackalloc byte[w * h * (_colorDepth == ColorDepth.ColourDepth65K ? 2 : 3)]; // span used to form the data to be written out to the SPI interface
+
+            // set the colourbyte array to represent the fill colour 
+            if (_colorDepth == ColorDepth.ColourDepth65K )
+            {
+                (colourBytes[0], colourBytes[1]) = Color565(color);
+            }
+            else
+            {
+                colourBytes[0] = (byte)((_colorSequence == ColorSequence.ABC ? color.B : color.R) >> 2);
```

Comment:
NIT: Should ColorSequence.ABC really be BGR?

Reply:
Well it was ABC/CBA because that is what the datasheet described but as the API is user facing I think that BGR/RGB is correct so I have updated the PR with this.